### ISSUE TITLE
`tm_plugin_init_i` is never called after the boot phase, even if a plugin is loaded for the first time

### ISSUE THREAD

**Issue by Selmar** [OurMachinery/themachinery-public#814]
I noticed this, because I always start with an empty project, and then navigate to my own project.

When closing the editor, `tm_plugin_shutdown_i` is still called for any plugin that never had its init function called.

**Comment by simon-ourmachinery** [comment]
`tm_plugin_init_i ` is only called during the initial load of the plugin. 
> tm_plugin_init_i is typically called as early as possible after all plugins have been loaded. tm_plugin_init_i is not called when a plugin is reloaded. [API Documentation](https://ourmachinery.com//apidoc/foundation/plugin_callbacks.h.html#tm_plugin_init_interface_name)

If you wamnted to cover reload cases you would need to make use of `tm_plugin_reload_i`

**Comment by simon-ourmachinery** [comment]
@niklas-ourmachinery I have investigated this a bit more. I miss understood the first message: What @Selmar refers to is that init, shutdown etc of a plugin asset won't be called.

Since there is no way to ask the api reg for plugin specific impls. I tried to implement a context-aware `context_implementations(context,name,version)` which I would call in `plugin_asset` after the loaded of plugins. The problem here is just that as soon as the "context" in the API registry is not the same as the plugin name my implementation does not work anymore. Which can happen when the user makes use of `begin_context / close_context`.
